core/modules/utils/csv.js

```diff
@@ -12,35 +12,109 @@ A barebones CSV parser
 /*global $tw: false */
 "use strict";
 
+var QUOTE = '"';
+
+var getCellInfo = function(text, start, length, SEPARATOR) {
+	var isCellQuoted = text.charAt(start) === QUOTE;
+	var cellStart = isCellQuoted ? start + 1 : start;
+	
+	if (text.charAt(i) === SEPARATOR) {
+		return [cellStart, cellStart, false];
+	}
+	
+	for (var i = cellStart; i < length; i++) {
+		var cellCharacter = text.charAt(i);
+		var isEOL = cellCharacter === "\n" || cellCharacter === "\r";
+		
+		if (isEOL && !isCellQuoted) {
+			return [cellStart, i, false];
+			
+		} else if (cellCharacter === SEPARATOR && !isCellQuoted) {
+			return [cellStart, i, false];
+			
+		} else if (cellCharacter === QUOTE && isCellQuoted) {
+			var nextCharacter = i + 1 < length ? text.charAt(i + 1) : '';
+			if (nextCharacter !== QUOTE) {
+				return [cellStart, i, true];
+			} else {
+				i++;
+			}
+		}
+	}
+	
+	return [cellStart, i, isCellQuoted];
+}
+	
+exports.parseCsvString = function(text, options) {
+	options = options || {};
+	var SEPARATOR = options.separator || ",",
```

Comment:
Maybe completely unnecessary, but you _could_ throw an error if `options.separator` contains a double quote, since that is a hard-coded, "magic" character in the parsing.

Reply:
Also, what happens if you use \r or \n as separators?

Reply:
>  you could throw an error
I feel like TW's way is to just fail gracefully rather than throw exceptions, and in that context the best option would probably be to just parse it that way and return whatever mess you get. I'll add a test case for that just to make sure no exception is thrown :).

Should also add a test case for empty input and such!